BrainPortal/app/models/cluster_task.rb

```
@@ -1687,6 +1689,151 @@ def docker_commands
 
 
   ##################################################################
+  # Internal Subtask Submission Mechanism
+  ##################################################################
+
+  public
+
+  # Handles +subtasks+ submitted while a task is running. To submit a subtask, a
+  # task must create a ".new-task-*.json" JSON file at the root of its
+  # work directory. Once a JSON file has been handled, it is deleted.
+  def submit_subtasks_from_json
+    workdir = self.full_cluster_workdir
+    return nil if workdir.blank? || ! File.directory?(workdir) # in case workdir doesn't exist yet
+    Dir.chdir(workdir) do
+      taskfiles = Dir.glob(".new-task-*.json")
+      return unless taskfiles.present?
+      current_status = self.status
+      return unless self.status_transition(current_status, "Subtasking") # ensures only one process can do the rest
+      taskfiles.each do |taskfile|
+        #self.addlog("Found new subtask file: #{taskfile}")
+        begin
+          file_content = File.read(taskfile)
+          self.submit_task_from_string(taskfile,file_content)
+        rescue => ex
+          self.addlog_exception(ex,"Error submitting subtask")
+        ensure
+          File.unlink(taskfile) rescue nil # to make sure we won't retry forever
+          #For debug
+          #File.rename(taskfile,"#{taskfile}.processed")
+        end
+      end # each new json descriptor
+      self.status_transition("Subtasking", current_status)
+    end
+  rescue  # TODO log error?
+    nil
```

May not be a very good idea to mask an exception...